_posts/2017-04-25-the-everchanging-software-process.md

```diff
@@ -0,0 +1,37 @@
+---
+layout: post
+title: The Everchanging Software Development Process
+author: mikkoronkkomaki
+excerpt: Why and how software development should be viewed as something that's under constant change.  
+tags:
+- software processes
+- methodologies
+- autonomy
+- agile retrospectives
+---
+
+Especially among us software developers, processes are some what of a curse word in many occasions. Personally I find that in most of those cases this repulsion is just. Mainly because what processes to many people working on the field mean, is something that basically just hinders everyday work, something that has been dictated from above and that the people with boots on the ground have little to none possibilities to affect.
+ 
+Then again in order to ease parts of the every day work and especially the co-operation with other people, we need at least some sort of structure. This is pretty much one of the key purpose of all the main methodologies and processes. There are tons of good books, blogs and articles on the issue so I'm not going to dwell on them for much longer and pretty much tell a real life story of how our team has approached the matter.
+ 
+In many cases, when starting off with a new project at Solita, we're not actually given that much as in terms of predicated means of how to conduct our every day work. Some things like tech stack might be restricted by the customers needs or environments. Other things like financial contracts will of course have an effect on e.g. deadlines work estimates etc. But pretty much the only underlying unspoken requirements that are "dictated from above" are: Serve the customer to the best of your abilities, in a way that's financially profitable and is not in contradiction with any legal or financial contract made with them. The rest is more or less left for the team to figure out.
```

Comment:
customers needs -> customer's needs

yleensä sopimuksen kohdalla ei sanota että "contradiction with" vaan "in violation of" jos viitataan varsinaisesti sopimusrikkomukseen, mutta tarkoitetaanko tässä oikeastaan sitä?   Voisiko tuon lopun ilmaista näin (jos ymmärsin oikein ajatuksen):
 
The unspoken rule is: Serve the customer to the best of your abilities, within the boundaries defined by the contract. It is left for the team to figure out how to actually do this.